File: CampingNeretva/CampingNeretva.Service/Database/Parcel.cs
﻿using System;
using System.Collections.Generic;

namespace CampingNeretva.Service.Database;

public partial class Parcel
{
    public int ParcelId { get; set; }

    public int ParcelNumber { get; set; }

    public bool Shade { get; set; }

    public bool Electricity { get; set; }

    public string? Description { get; set; }

    public bool AvailabilityStatus { get; set; }

    public int? ParcelAccommodationId { get; set; }

    public int? ParcelTypeId { get; set; }

    public virtual ParcelAccommodation? ParcelAccommodation { get; set; }

    public virtual ICollection<ParcelImage> ParcelImages { get; set; } = new List<ParcelImage>();

    public virtual ParcelType? ParcelType { get; set; }

    public virtual ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
}

File: CampingNeretva/UI/campingneretva_mobile/lib/models/parcel.dart
class Parcel {
  final int parcelId;
  final int parcelNumber;
  final bool shade;
  final bool electricity;
  final bool availabilityStatus;
  final List<String> images;

  Parcel({
    required this.parcelId,
    required this.parcelNumber,
    required this.shade,
    required this.electricity,
    required this.availabilityStatus,
    required this.images,
  });

  factory Parcel.fromJson(Map<String, dynamic> json) {
    return Parcel(
      parcelId: json['parcelId'],
      parcelNumber: json['parcelNumber'],
      shade: json['shade'],
      electricity: json['electricity'],
      availabilityStatus: json['availabilityStatus'],
      images: List<String>.from(json['images'].map((img) => img['url'])),
    );
  }
}
